catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroSettingsActivityTest.java

```diff
@@ -0,0 +1,175 @@
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroMainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroMainMenuActivity.class, true, false);
+
+	private List<String> allSettings = new ArrayList<>(Arrays.asList(SETTINGS_SHOW_PHIRO_BRICKS, PHIRO_INITIALIZED));
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+
+	@Before
+	public void setUp() {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+		setAllSettingsTo(false);
+
+		baseActivityTestRule.launchActivity(null);
+
+		idlingResource = baseActivityTestRule.getActivity().getIdlingResource();
+		Espresso.registerIdlingResources(idlingResource);
+	}
+
+	@After
+	public void tearDown() {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+		for (String setting : initialSettings.keySet()) {
+			sharedPreferencesEditor.putBoolean(setting, initialSettings.get(setting));
+		}
+		sharedPreferencesEditor.commit();
+		initialSettings.clear();
+		Espresso.unregisterIdlingResources(idlingResource);
+	}
+
+	private void setAllSettingsTo(boolean value) {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+
+		for (String setting : allSettings) {
+			sharedPreferencesEditor.putBoolean(setting, value);
+		}
+		sharedPreferencesEditor.commit();
+	}
+
+	@Test
+	public void phiroIsInitiallyActivatedTest() {
+		openSettings();
+
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		assertTrue("Phiro preference was not initially enabled!", sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
+	}
+
+	@Test
+	public void phiroIsFirstVisibleSettingTest() {
+		openSettings();
+
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks))
+				.atPosition(0)
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void phiroSettingsTest() {
+		openSettings();
+
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks))
+				.perform(click());
+
+		checkPreference(R.string.preference_title_enable_phiro_bricks, SETTINGS_SHOW_PHIRO_BRICKS);
+
+		String phiroLink = UiTestUtils.getResourcesString(R.string.phiro_preference_link);
+
+		Intents.init();
+		Matcher<Intent> expectedIntent = allOf(hasAction(Intent.ACTION_VIEW), hasData(phiroLink));
+		intending(expectedIntent).respondWith(new Instrumentation.ActivityResult(0, null));
+		onData(PreferenceMatchers.withTitle(R.string.phiro_preference_title)).perform(click());
+		intended(expectedIntent);
+		Intents.release();
+	}
+
+	private void openSettings() {
```

Comment:
use BaseActivityInstrumentationRule<SettingsActivity> to open directly the activity under test.
DO NOT use the ui to navigate to the activity that you want to test. this practice was one of the main reasons the old ui test setup was slow as fuck and flaky.

please have a look at:
https://confluence.catrob.at/display/CAT/Espresso+testing+-+How+to
theres a whole paragraph about this "Open the activity you want to test directly"